WordPress/Classes/ViewRelated/Blog/SiteManagement/TableViewHeaderDetailView.swift

```diff
@@ -0,0 +1,184 @@
+                titleLabel.preferredMaxLayoutWidth = labelWidth
+                detailLabel.preferredMaxLayoutWidth = labelWidth
+                
+                stackWidthConstraint?.constant = layoutWidth
+                setNeedsUpdateConstraints()
+            }
+        }
+    }
+    
+    // MARK: - Private Aliases
+    
+    private typealias Style = WPStyleGuide.TableViewHeaderDetailView
+
+    // MARK: - Private Properties
+
+    private let titleLabel: UILabel = {
+        let titleLabel = UILabel()
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+        titleLabel.numberOfLines = 0
+        titleLabel.lineBreakMode = .ByWordWrapping
+        titleLabel.font = Style.titleFont
+        titleLabel.textColor = Style.titleColor
+        
+        return titleLabel
+    }()
+    
+    private let detailLabel: UILabel = {
+        let detailLabel = UILabel()
+        detailLabel.translatesAutoresizingMaskIntoConstraints = false
+        detailLabel.numberOfLines = 0
+        detailLabel.lineBreakMode = .ByWordWrapping
+        detailLabel.font = Style.detailFont
+        detailLabel.textColor = Style.detailColor
+        
+        return detailLabel
+    }()
+    
+    private let stackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [])
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .Vertical
+        stackView.alignment = .Fill
+        stackView.distribution = .Fill
+        stackView.spacing = Style.headerDetailSpacing
+        stackView.layoutMargins = Style.layoutMargins
+        stackView.layoutMarginsRelativeArrangement = true
+        
+        return stackView
+    }()
+    
+    private var stackWidthConstraint: NSLayoutConstraint?
+
+    // MARK: - Initializers
+    
+    /// Convenience initializer for TableViewHeaderDetailView
+    ///
+    /// - Parameters:
+    ///     - title: String displayed in standard section header style
+    ///     - detail: String displayed in standard section footer style
+    ///
+    convenience public init(title: String?, detail: String?) {
+        self.init(reuseIdentifier: nil)
+        defer {
+            self.title = title ?? ""
+            self.detail = detail ?? ""
+        }
+    }
+    
+    override public init(reuseIdentifier: String?) {
+        super.init(reuseIdentifier: reuseIdentifier)
+        stackSubviews()
+    }
+    
+    required public init?(coder aDecoder: NSCoder){
+        super.init(coder: aDecoder)
+        stackSubviews()
+    }
+    
+    private func stackSubviews() {
+        contentView.addSubview(stackView)
+        stackView.addArrangedSubview(titleLabel)
+        stackView.addArrangedSubview(detailLabel)
+        
+        stackView.centerXAnchor.constraintEqualToAnchor(contentView.centerXAnchor).active = true
+        stackWidthConstraint = stackView.widthAnchor.constraintEqualToConstant(UIScreen.mainScreen().bounds.width)
```

Comment:
However, wouldn't the constant for the `stackWidthConstraint` not be updated and still have the original screen width value, rather than the new width value?